### PR TITLE
Create data_bag before uploading the items

### PR DESCRIPTION
**Pull request by rsalevsky**
this was missing from the last PR

### CODE REVIEW

File: crowbar_framework/lib/crowbar/chef/upload.rb [hunk 1]
```diff
@@ -69,6 +69,15 @@ def data_bags
           loader = ::Chef::Knife::Core::ObjectLoader.new(::Chef::DataBagItem, logger)
           data_bags = loader.find_all_object_dirs(chef_data_bags_path) || []
           data_bags.each do |data_bag|
+            begin
+              api.post_rest("data", name: data_bag)
+            rescue Net::HTTPServerException => e
+              unless e.response.code == 409
```
Comment (MaximilianMeister) [inline crowbar_framework/lib/crowbar/chef/upload.rb]: not sure but shouldn't it be an `if` instead of an `unless`? or why do you skip when you get 409?

Comment (rsalevsky) [inline crowbar_framework/lib/crowbar/chef/upload.rb]: 409 is the response code if the data_bag already exists

**Comment by MaximilianMeister** [comment]
> 409 is the response code if the data_bag already exists

okay, then maybe it wouldn't hurt to have at least a log message in `production.log`, so sth like

```ruby
            begin
              logger.info("Creating data_bag #{data_bag}...")
              api.post_rest("data", name: data_bag)
            rescue Net::HTTPServerException => e
              if e.response.code == "409"
                logger.info("data_bag #{data_bag} already exists: #{e.message}")
              else
                logger.error("Creating data_bag #{data_bag} failed (#{e.response.code})")
                return false
              end
            end
```

**Comment by MaximilianMeister** [comment]
+1

**Comment by jdsn** [comment]
:+1:

**Pull request merged**